From: Artur Rona <ari-tczew@ubuntu.com>
Description: Fix FTBFS with ld --no-add-needed. Remove unnecessary dependency on libva for libva-x11. libva-x11 uses none of libva's symbols. But it uses -lX11, which fixes FTBFS on ubuntu natty.
Bug-Debian: http://bugs.debian.org/609554
Author: Matthias Klose <doko@ubuntu.com>
Last-Update: 2011-02-13

--- a/va/Makefile.am
+++ b/va/Makefile.am
@@ -71,8 +71,8 @@ SUBDIRS				+= x11
 lib_LTLIBRARIES			+= libva-x11.la
 libva_x11_la_SOURCES		= 
 libva_x11_la_LDFLAGS		= $(LDADD)
-libva_x11_la_DEPENDENCIES	= $(libvacorelib) x11/libva_x11.la
-libva_x11_la_LIBADD		= $(libvacorelib) x11/libva_x11.la \
+libva_x11_la_DEPENDENCIES	= x11/libva_x11.la
+libva_x11_la_LIBADD		= x11/libva_x11.la \
 	$(LIBVA_LIBS) $(X11_LIBS) $(XEXT_LIBS) $(DRM_LIBS) $(XFIXES_LIBS) -ldl
 
 if USE_GLX
--- a/test/basic/Makefile.am
+++ b/test/basic/Makefile.am
@@ -25,7 +25,8 @@ check_PROGRAMS = test_01 test_02 test_03
 
 AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
-TEST_LIBS = $(top_builddir)/va/$(libvabackendlib)
+TEST_LIBS = $(top_builddir)/va/$(libvabackendlib) \
+  $(top_builddir)/va/$(libvacorelib) $(X11_LIBS)
 
 test_01_LDADD = $(TEST_LIBS)
 test_01_SOURCES = test_01.c
